fix windows-hide-console for bun SFE

rithvikvibhu opened the pull request:
### What does this PR do?

<!-- **Please explain what your changes do**, example: -->

#15894 added an option `--windows-hide-console` but it isn't really called. This PR fixes it.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

before, generated exe always shows as `console` even with `windows-hide-console` set:
```
$ file testbin.exe
testbin.exe: PE32+ executable for MS Windows 6.00 (console), x86-64, 13 sections
```

now, if the flag is set:
```
$ file testbin.exe
testbin.exe: PE32+ executable for MS Windows 6.00 (GUI), x86-64, 13 sections
```

cirospaciari commented on src/windows.zig:
```diff
@@ -3648,10 +3648,10 @@ pub const Subsystem = enum(u16) {
 pub fn editWin32BinarySubsystem(fd: bun.sys.File, subsystem: Subsystem) !void {
     comptime bun.assert(bun.Environment.isWindows);
-    if (bun.windows.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
+    if (kernel32.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
```
is this change necessary? bun.windows should do the same

rithvikvibhu replied:
`bun.windows` didn't work for me? Which is why I changed to kernel32.

```
src\windows.zig:3651:20: error: root source file struct 'windows' has no member named 'SetFilePointerEx'
    if (bun.windows.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
        ~~~~~~~~~~~^~~~~~~~~~~~~~~~~
src\windows.zig:1:1: note: struct declared here
//! Platform specific APIs for Windows
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    inject: src\StandaloneModuleGraph.zig:754:49
```

cirospaciari reviewed (changes requested):
Thank you for the contributions, just some comments + can you add a test?

cirospaciari commented on src/windows.zig:
```diff
@@ -3648,10 +3648,10 @@ pub const Subsystem = enum(u16) {
 pub fn editWin32BinarySubsystem(fd: bun.sys.File, subsystem: Subsystem) !void {
     comptime bun.assert(bun.Environment.isWindows);
-    if (bun.windows.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
+    if (kernel32.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
```
```suggestion
    if (bun.windows.SetFilePointerEx(fd.handle.cast(), pe_header_offset_location, null, std.os.windows.FILE_BEGIN) == 0)
```

rithvikvibhu commented:
@cirospaciari replied to the comments. For the test, I think `test\bundler\bundler_compile.test.ts` is a good place for it. But how do you think it should test since it directly modifies the exeutable? Run `file`? Or do whatever file does (read the exe and ensure the bytes are changed?

rithvikvibhu commented on test/bundler/bundler_compile.test.ts:
```diff
@@ -626,4 +626,37 @@ error: Hello World`,
   });
+  // This test is only for Windows
+  if (isWindows) {
+    itBundled.only("compile/WindowsHideConsole", {
```
d'oh forgot about that sorry. Removed it.

cirospaciari replied:
Thanks will let CI run and see :D

rithvikvibhu commented:
fyi none of the failing tests are related to these changes afaics

rithvikvibhu commented:
Hey @cirospaciari! I saw that `PEFile` was added a few days ago that does proper parsing of the exe.

I've updated from main again and modified to set the subsystem value with PEFile. The added test passes. FYI: there are other tests that fail on windows, but aren't related to any change this PR makes.

Lemme know if something needs to change (or feel free to push commits directly).

SunsetTechuila commented:
except https://github.com/oven-sh/bun/pull/20594/commits/7ea488654b719d8c1d7542a1af66048da2ed802b this pr is the same as https://github.com/oven-sh/bun/pull/20530

SunsetTechuila commented:
I' ve updated my PR
